includes/Modules/Search_Console.php

```diff
@@ -418,18 +403,20 @@ function( $dataset ) {
 			);
 
 			$has_data = false;
-			foreach ( $responses as $key => $response ) {
-				if ( is_wp_error( $response ) || ! is_array( $response ) || empty( $response ) || ! isset( $response[0] ) ) {
+
+			// Go thru results, any impressions means the URL has data.
+			foreach ( $responses['sc-site-analytics'] as $key => $response ) {
+				if ( is_wp_error( $response ) || empty( $response ) || ! isset( $response ) ) {
 					continue;
 				}
 
-				if ( $response[0]->clicks > 0 || $response[0]->impressions > 0 ) {
+				if ( $response->impressions > 0 ) {
 					$has_data = true;
 					break;
 				}
 			}
 
-			set_transient( $transient_key, (int) $has_data, 2 * HOUR_IN_SECONDS );
+			set_transient( $transient_key, (int) $has_data, $has_data ? DAY_IN_SECONDS : HOUR_IN_SECONDS );
```

Comment:
Why a different expiration here?

Reply:
when we find data, we really don't need to check for data again for some time (a day). if we don't find data, we can check again more frequently, maybe this is a new site? (1 hour)